include/deal.II/tests/tests.h

```diff
@@ -0,0 +1,149 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2021 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//---------------------------------------------------------------
+
+#ifndef dealii_tests_h
```

Comment:
I have move the classes related to logging into the new file. We can add additional functions here depending on the feedback of users.

Reply:
@peterrum: I think in addition the header <fstream> and the definition `void cat_file(const char *filename)` need to be moved into this file. Otherwise, one gets a build error as `cat_file` is called in `struct MPILogInitAll`.

Reply:
Thanks for the feedback! I'll add it!